src/main/java/com/list/todo/controllers/TasksController.java

```diff
@@ -3,59 +3,141 @@
 import java.util.List;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.list.todo.entity.Project;
 import com.list.todo.entity.Task;
-import com.list.todo.services.ProjectService;
+import com.list.todo.entity.TodoList;
+import com.list.todo.security.CurrentUser;
+import com.list.todo.security.UserPrincipal;
 import com.list.todo.services.TaskService;
+import com.list.todo.services.TodoListService;
 
 @RestController
+@RequestMapping("/api/todolists/{todoListId}/tasks")
```

Comment:
Think about enriching todolist controller with task support because compound id's will make you cry.